File: ios/Classes/Models/HMSSimulcastSettingsExtension.swift
//
//  HMSSimulcastSettingsExtension.swift
//  hmssdk_flutter
//
//  Created by Govind on 06/12/22.
//

import Foundation
import HMSSDK

class HMSSimulcastSettingsExtension {
    static func toDictionary(_ simulcast: HMSSimulcastSettings) -> [String:Any] {
        var dict = [String:Any]()
        
        if let video = simulcast.video {
            dict["video"] = HMSSimulcastSettingsPolicyExtension.toDictionary(video)
        }
        
        if let screen = simulcast.screen {
            dict["screen"] = HMSSimulcastSettingsPolicyExtension.toDictionary(screen)
        }
        
        return dict
    }
}

File: android/src/main/kotlin/live/hms/hmssdk_flutter/HMSSimulcastSettingsExtension.kt
package live.hms.hmssdk_flutter

import live.hms.video.media.settings.HMSSimulcastLayerDefinition
import live.hms.video.sdk.models.role.Simulcast

class HMSSimulcastSettingsExtension {
    companion object {
        fun toDictionary(simulcast: Simulcast): HashMap<String, Any?> {
            val hashMap = HashMap<String, Any?>()

            if(simulcast.video != null){
                hashMap["video"] = HMSSimulcastSettingsPolicyExtension.toDictionary(simulcast.video!!)
            }
            if(simulcast.screen != null){
                hashMap["screen"] = HMSSimulcastSettingsPolicyExtension.toDictionary(simulcast.screen!!)
            }
            return hashMap
        }
    }
}